openshift-environment-driver/src/main/java/org/jboss/pnc/environment/openshift/OpenshiftStartedEnvironment.java

```diff
@@ -627,10 +567,12 @@ public String getId() {
     public void cancel() {
         cancelRequested = true;
 
-        creatingPod.ifPresent(f -> f.cancel(false));
-        creatingService.ifPresent(f -> f.cancel(false));
+        creatingPod.cancel(false);
+        creatingService.cancel(false);
         creatingRoute.ifPresent(f -> f.cancel(false));
 
+        cancelAndClearMonitors();
```

Comment:
From the changes I'm seeing, it cleans up the old code quite a lot in regards on how cancel and error handling is done. +1 for me!

Reply:
Thanks for the comments, all accepted.